pkg/api/admin/openshiftcluster_validatestatic.go

```diff
@@ -29,7 +29,7 @@ func (sv openShiftClusterStaticValidator) validateDelta(oc, current *OpenShiftCl
 		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodePropertyChangeNotAllowed, err.Target, err.Message)
 	}
 
-	if !(oc.Properties.MaintenanceTask == "" || oc.Properties.MaintenanceTask == MaintenanceTaskEverything || oc.Properties.MaintenanceTask == MaintenanceTaskOperator) {
+	if !(oc.Properties.MaintenanceTask == "" || oc.Properties.MaintenanceTask == MaintenanceTaskEverything || oc.Properties.MaintenanceTask == MaintenanceTaskOperator || oc.Properties.MaintenanceTask == MaintenanceTaskRenewCerts) {
```

Comment:
If this list is growing, should we format this more like an enum (with consts and then a slice we loop through)? @ross-bryan thoughts?

Reply:
big +1, this conditional is now up to 4 items, and increases cognitive load and reduces readability in its current state. We could add a helper function to return a list of "valid" `MaintenanceTask`'s and verify that our valid list `contains` _this_ `oc.Properties.MaintenanceTask`.

In go1.18 `slices.Contains` ships and we can stop writing our own helper funcs

Reply:
I think we can implement the slices.Contains method once we start using 1.18 version.
`if !slices.Contains([]string{"", MaintenanceTaskEverything, MaintenanceTaskOperator, MaintenanceTaskRenewCerts}, oc.Properties.MaintenanceTask) {
		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, "properties.maintenanceTask", "Invalid enum parameter.")
	}`

Reply:
```suggestion
	if !(oc.Properties.MaintenanceTask == "" || oc.Properties.MaintenanceTask == MaintenanceTaskEverything || oc.Properties.MaintenanceTask == MaintenanceTaskOperator || oc.Properties.MaintenanceTask == MaintenanceTaskRenewCerts) {
```

Reply:
if !slices.Contains([]string{"", MaintenanceTaskEverything, MaintenanceTaskOperator, MaintenanceTaskRenewCerts}, oc.Properties.MaintenanceTask) {
		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, "properties.maintenanceTask", "Invalid enum parameter.")
	}

Reply:
if !slices.Contains([]string{"", MaintenanceTaskEverything, MaintenanceTaskOperator, MaintenanceTaskRenewCerts}, oc.Properties.MaintenanceTask) {
		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, "properties.maintenanceTask", "Invalid enum parameter.")
	}